netneurotools/stats.py

```diff
@@ -784,3 +785,148 @@ def gen_spinsamples(coords, hemiid, n_rotate=1000, check_duplicates=True,
         return spinsamples, cost
 
     return spinsamples
+
+
+def get_dominance_stats(X, y, use_adjusted_r_sq=True, verbose=False):
+    """
+    Returns the dominance analysis statistics for multilinear regression.
+
+    This is a rewritten & simplified version of [DA1]_. It is briefly
+    tested against the original package, but still in early stages.
+    Please feel free to report any bugs.
+
+    Warning: Still work-in-progress. Parameters might change!
+
+    Parameters
+    ----------
+    X : (N, M) array_like
+        Input data
+    y : (N,) array_like
+        Target values
+    use_adjusted_r_sq : bool, optional
+        Whether to use adjusted r squares. Default: True
+    verbose : bool, optional
+        Whether to print debug messages. Default: False
+
+    Returns
+    -------
+    model_metrics : dict
+        The dominance metrics, currently containing `individual_dominance`,
+        `partial_dominance`, `total_dominance`, and `full_r_sq`.
+    model_r_sq : dict
+        Contains all model r squares
+
+    Notes
+    -----
+    Example usage
+
+    .. code:: python
+
+        from netneurotools.stats import get_dominance_stats
+        from sklearn.datasets import load_boston
+        X, y = load_boston(return_X_y=True)
+        model_metrics, model_r_sq = get_dominance_stats(X, y)
+
+    To compare with [DA1]_, use `use_adjusted_r_sq=False`
+
+    .. code:: python
+
+        from dominance_analysis import Dominance_Datasets
+        from dominance_analysis import Dominance
+        boston_dataset=Dominance_Datasets.get_boston()
+        dominance_regression=Dominance(data=boston_dataset,
+                                       target='House_Price',objective=1)
+        incr_variable_rsquare=dominance_regression.incremental_rsquare()
+        dominance_regression.dominance_stats()
+
+    References
+    ----------
+    .. [DA1] https://github.com/dominance-analysis/dominance-analysis
+
+    """
+
+    from itertools import combinations
+    from sklearn.linear_model import LinearRegression
+
+    # sklearn linear regression wrapper
+    def get_reg_r_sq(X, y):
+        lin_reg = LinearRegression()
+        lin_reg.fit(X, y)
+        yhat = lin_reg.predict(X)
+        SS_Residual = sum((y - yhat) ** 2)
+        SS_Total = sum((y - np.mean(y)) ** 2)
+        r_squared = 1 - (float(SS_Residual)) / SS_Total
+        adjusted_r_squared = 1 - (1 - r_squared) * \
+            (len(y) - 1) / (len(y) - X.shape[1] - 1)
+        if use_adjusted_r_sq:
+            return adjusted_r_squared
+        else:
+            return r_squared
+
+    def complete_model_rsquare(X, y):
+        return get_reg_r_sq(X, y)
```

Comment:
Isn't this redundant? Can you not get rid of `complete_model_rsquare()` and just call `get_reg_r_sq()` instead?